### PR TITLE
Implement ConceptMap translate operation

### LINKED ISSUE #3344: Implement ConceptMap $translate

### PR DESCRIPTION
**Pull request by mattwiller**
Adding the `$translate` operation on `ConceptMap` resources, available either as:
- `POST ConceptMap/:id/$translate` to translate using a given ConceptMap
- `POST ConceptMap/$translate` to use the `url` or `source` parameters to look up the ConceptMap

Currently, only simple enumerated mappings are supported via `ConceptMap.group`

Resolves #3344

### CODE REVIEW

File: packages/server/src/fhir/operations/README.md [hunk 1]
```diff
@@ -57,7 +58,7 @@ interface ProjectInitParameters {
   ownerEmail?: string;
 }
 
-export function projectInitHandler(req: Request, res: Response): void {
+export async function projectInitHandler(req: Request, res: Response): Promise<void> {
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/README.md:61]: Does this need to be `async`?

File: packages/server/src/fhir/operations/conceptmaptranslate.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import { Operator, allOk, badRequest, notFound } from '@medplum/core';
+import { Request, Response } from 'express';
+import { getOperationDefinition } from './definitions';
+import { CodeableConcept, Coding, ConceptMap, ConceptMapGroup, OperationOutcome } from '@medplum/fhirtypes';
+import { getAuthenticatedContext } from '../../context';
+import { sendOutcome } from '../outcomes';
+
+const operation = getOperationDefinition('ConceptMap', 'translate');
+
+type ConceptMapTranslateParameters = {
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/conceptmaptranslate.ts:11]: Curious why `type` over `interface`?

Not a big deal, but hopefully as a team we can formulate a framework on how to choose.

In general, I've defaulted to `interface` in the past based on recommendations from the TypeScript's page on performance: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

File: packages/server/src/fhir/operations/conceptmaptranslate.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+};
+
+export async function conceptMapTranslateHandler(req: Request, res: Response): Promise<void> {
+  const params = parseInputParameters<ConceptMapTranslateParameters>(operation, req);
+
+  const map = await lookupConceptMap(params, req.params.id);
+  if (isOutcome(map)) {
+    sendOutcome(res, map);
+    return;
+  } else if (!map.group) {
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/conceptmaptranslate.ts:39]: Another super minor thing, just for aligning coding styles, is using `else` when the parent block uses `return`  Functionally, totally equivalent, so this is purely a style question.  It'd just be nice to be consistent throughout the code base.

File: packages/server/src/fhir/operations/conceptmaptranslate.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    sendOutcome(res, badRequest('ConceptMap does not specify a mapping group', 'ConceptMap.group'));
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/conceptmaptranslate.ts:40]: Excellent use of 2nd arg to `badRequest()`! 👍

**Review by codyebberson — APPROVED** [review]
Nice, excellent work @mattwiller - this looks great.

A handful of style questions, nothing serious or material.  The main motivation of the questions is to align on styles and philosophies so we can maintain a consistent feel.

File: packages/server/src/fhir/operations/README.md (hunk 1 above)
Comment (mattwiller) [reply packages/server/src/fhir/operations/README.md:61]: Yep, it uses `await` on line 68

File: packages/server/src/fhir/operations/conceptmaptranslate.ts (hunk 3 above)
Comment (mattwiller) [reply packages/server/src/fhir/operations/conceptmaptranslate.ts:39]: I generally favor an early-return style for error cases, leaving the main code path unindented to highlight the "happy path" when reading the code — hence no `else` here.  Thoughts on that general idea?

File: packages/server/src/fhir/operations/conceptmaptranslate.ts (hunk 2 above)
Comment (mattwiller) [reply packages/server/src/fhir/operations/conceptmaptranslate.ts:11]: I was going off of [this line of thinking](https://blog.logrocket.com/types-vs-interfaces-typescript/#when-use-types-vs-interfaces), that `type` is more expressive and in theory works exactly the same for simple cases like this one.  I see the performance argument, but my read was that it only mattered when doing definition merging or extending, which isn't generally a major use case from what I've seen

File: packages/server/src/fhir/operations/conceptmaptranslate.ts (hunk 3 above)
Comment (mattwiller) [reply packages/server/src/fhir/operations/conceptmaptranslate.ts:39]: Ah, I see — I do like to group error cases into an `if`/`else if` as well, to visually cluster them together and further highlight the happy path.  If you'd prefer it the other way, we could enforce that style with https://eslint.org/docs/latest/rules/no-else-return by setting `allowElseIf: false`

**Pull request merged**